discord/widget.py

```diff
@@ -204,8 +204,10 @@ def __init__(self, *, state, data):
         channels = {channel.id: channel for channel in self.channels}
         for member in data.get('members', []):
             connected_channel = _get_as_snowflake(member, 'channel_id')
-            if connected_channel:
+            if connected_channel in channels:
                 connected_channel = channels[connected_channel]
+            elif connected_channel:
+                connected_channel = WidgetChannel(id=connected_channel, name=None, position=None)
```

Comment:
Probably better to mock these so they don't change the typing. The current implementation as-is currently fails since `__str__` will return `None`.

Reply:
Alright, I made the requested changes.